kotlinx-coroutines-core/jvm/test/channels/ChannelLFStressTest.kt

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2016-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.coroutines.channels
+
+import kotlinx.atomicfu.*
+import kotlinx.coroutines.*
+import java.util.concurrent.atomic.AtomicLong
+import java.util.concurrent.atomic.AtomicLongArray
+import kotlin.math.*
+import kotlin.test.*
+
+/**
+ * Tests lock-freedom of send and receive operations on rendezvous channel.
+ * There is a single channel with two sender and two receiver threads.
+ * When one sender or receiver gets suspended at most one other operation is allowed to cease having progress
+ * (`allowSuspendedThreads = 1`).
+ *
+ * **Note**: In the current implementation buffered channels are not lock-free, so this test would fail
+ * *         if channel is created with a buffer.
+ */
+class ChannelLFStressTest : TestBase() {
+    private val nSeconds = 5 * stressTestMultiplier
+    private val env = LockFreedomTestEnvironment("RendezvousChannelStressLFTest", allowSuspendedThreads = 1)
+    private lateinit var channel: Channel<Long>
+
+    private val sendIndex = AtomicLong()
+    private val receiveCount = AtomicLong()
+    private val duplicateCount = AtomicLong()
+
+    private val nCheckedSize = 10_000_000
+    private val nChecked = (nCheckedSize * Long.SIZE_BITS).toLong()
+    private val receivedBits = AtomicLongArray(nCheckedSize) // bit set of received values
+
+    @Test
+    fun testRendezvousLockFreedom() {
```

Comment:
When I run tests in a batch, I consistently see the following message:
```
--- 0: Performed 0 operations
Sent: 2, Received: 0, dups: 0
```
Isn't it an indicator of some races in LFTE machinery?

Reply:
It looks  ok. It started sending, but have not updated the number of sent operation yet.